lib/msal-browser/docs/navigation.md

```diff
@@ -0,0 +1,92 @@
+# Intercepting or Overriding Window Navigation
+
+By default `msal-browser` uses `window.location.assign` and `window.location.replace` to redirect your application to external urls, such as the sign-in or sign-out pages, and internal urls, other pages in your app after returning from an external redirect. There may be situations, however, where you would like to override the default behavior with your own. For example, frameworks like React and Angular provide different APIs to handle client-side navigation without reloading the entire SPA. `msal-browser` exposes both the `INavigationClient` interface as well as the `NavigationClient` default implementation for you to provide your own custom implementation.
```

Comment:
Might be good to call out that navigating to another domain use navigateInternal may break your auth. And any other potential pitfalls with overriding this.

Reply:
Msal provides the url that should be navigated to so unless someone ignores it and tries to go to a different url this shouldn't happen. But I've included the disclaimer just in case.